components/TitleBar/src/TitleBar.WASDK.cs

```diff
@@ -0,0 +1,208 @@
+
+        Window.AppWindow.TitleBar.ExtendsContentIntoTitleBar = false;
+        this.Window.Activated -= Window_Activated;
+        SizeChanged -= this.TitleBar_SizeChanged;
+        Window.AppWindow.TitleBar.ResetToDefault();
+    }
+
+    private void UpdateRegionToSize()
+    {
+        // Update drag region if the size of the title bar changes.
+        SetDragRegionForCustomTitleBar();
+    }
+
+    private void Window_Activated(object sender, WindowActivatedEventArgs args)
+    {
+        if (args.WindowActivationState == WindowActivationState.Deactivated)
+        {
+            VisualStateManager.GoToState(this, WindowDeactivatedState, true);
+        }
+        else
+        {
+            VisualStateManager.GoToState(this, WindowActivatedState, true);
+        }
+    }
+
+    private void SetDragRegionForCustomTitleBar()
+    {
+        if (Window != null)
+        {
+            double scaleAdjustment = GetScaleAdjustment();
+
+            PART_RightPaddingColumn!.Width = new GridLength(Window.AppWindow.TitleBar.RightInset / scaleAdjustment);
+            PART_LeftPaddingColumn!.Width = new GridLength(Window.AppWindow.TitleBar.LeftInset / scaleAdjustment);
+
+            List<Windows.Graphics.RectInt32> dragRectsList = new();
+
+            Windows.Graphics.RectInt32 dragRectL;
+            dragRectL.X = (int)((PART_LeftPaddingColumn.ActualWidth
+                            + PART_ButtonsHolderColumn!.ActualWidth)
+                            * scaleAdjustment);
+            dragRectL.Y = 0;
+            dragRectL.Height = (int)(this.ActualHeight * scaleAdjustment);
+            dragRectL.Width = (int)((PART_IconColumn!.ActualWidth
+                                + PART_TitleColumn!.ActualWidth
+                                + PART_LeftDragColumn!.ActualWidth)
+                                * scaleAdjustment);
+            dragRectsList.Add(dragRectL);
+
+            Windows.Graphics.RectInt32 dragRectR;
+            dragRectR.X = (int)((PART_LeftPaddingColumn.ActualWidth
+                                + PART_IconColumn.ActualWidth
+                                + PART_ButtonsHolderColumn!.ActualWidth
+                                + PART_TitleHolder!.ActualWidth
+                                + PART_LeftDragColumn.ActualWidth
+                                + PART_ContentColumn!.ActualWidth)
+                                * scaleAdjustment);
+            dragRectR.Y = 0;
+            dragRectR.Height = (int)(this.ActualHeight * scaleAdjustment);
+            dragRectR.Width = (int)(PART_RightDragColumn!.ActualWidth * scaleAdjustment);
+            dragRectsList.Add(dragRectR);
+
+            Windows.Graphics.RectInt32[] dragRects = dragRectsList.ToArray();
+
+            Window.AppWindow.TitleBar.SetDragRectangles(dragRects);
+        }
+    }
+
+    [DllImport("Shcore.dll", SetLastError = true)]
+    internal static extern int GetDpiForMonitor(IntPtr hmonitor, Monitor_DPI_Type dpiType, out uint dpiX, out uint dpiY);
```

Comment:
Seems silly for a single method, but can use [CsWin32](https://github.com/microsoft/CsWin32) to generate these instead...